relay/sources/rudder-pkg/lib/rudder-pkg/rudderPkgUtils.py

```diff
@@ -356,28 +356,28 @@ def package_check(metadata, version):
     # do not compare with exiting version to allow people to reinstall or downgrade
     return name in DB['plugins']
 
-
-def check_plugin_compatibility(metadata, version):
-    # check that the given version is compatible with Rudder one
+def extract_plugin_version(metadata):
     match = re.match(
         r'((\d+\.\d+)(\.\d+(~(alpha|beta|rc)\d*)?)?)(-SNAPSHOT)?-(\d+)\.(\d+)', metadata['version']
     )
     if not match:
         fail('Invalid package version ' + metadata['version'])
-    rudder_version = match.group(1).replace('~alpha', 'a').replace('~beta', 'b').replace('~rc', 'c')
-    rudder_major = match.group(2)
-    major_version = match.group(7)
-    minor_version = match.group(8)
-    if rudder_major != RUDDER_MAJOR:
-        return False
-    if StrictVersion(rudder_version) > StrictVersion(RUDDER_VERSION):
-        return False
-
-    # check specific constraints
-    full_name = metadata['name'] + '-' + metadata['version']
-    if full_name in COMPATIBILITY_DB['incompatibles']:
-        return False
-    return True
+    return match
+
+def check_exact_compatibility(plugin_version):
+    # check that the given version is exactly the same as Rudder one
+    rudder_version = plugin_version.group(1).replace('~alpha', 'a').replace('~beta', 'b').replace('~rc', 'c')
+    return rudder_version == RUDDER_VERSION
+
+def check_loose_compatibility(plugin_version):
+    # check that the given version is compatible with Rudder minor one
+    rudder_minor = plugin_version.group(2)
+    return rudder_minor == RUDDER_MINOR
```

Comment:
I had a minute of "wait, does that means it doesn't compare major version", but it's actually ok. Perhaps update the commen to `... with Rudder major.minor one` to better convey the intent